app/routes/goal.py

```diff
@@ -0,0 +1,78 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request
+from .validation import get_model_from_id
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods = ["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    goals_response = [goal.to_dict() for goal in goals]
+    
+    return jsonify(goals_response), 200
+
+@goals_bp.route("<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    choosen_goal = get_model_from_id(Goal, goal_id)
+    
+    return jsonify({"goal": choosen_goal.to_dict()}), 200
+
+@goals_bp.route("", methods = ["POST"])
+def create_one_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError:
+        return jsonify({"details": "Invalid data"}), 400
```

Comment:
nice use of a try-except, but it would be great to include details about what data is invalid.